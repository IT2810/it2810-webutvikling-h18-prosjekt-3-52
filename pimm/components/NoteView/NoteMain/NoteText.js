import React from 'react';
import { Text, TextInput, View } from 'react-native';

export default class Title extends React.Component{
	constructor(props){
		super(props);
		this.state={
			note: "",
		};
		
	}
	
	changeNote(newNote){
		this.setState({note: newNote});
	
	}
	
	render() {
		return(
			<View style={{ flex: 1, alignItems: 'center', justifyContent: 'center', padding: 5 }}>
			<TextInput 
			placeholder="Write note here..."
			multiline = {true}
			//numberOfLines = {6}
			style={{padding: 5, width: 280, backgroundColor: 'beige'}}
			onChangeText={(note) => this.changeNote(note)}
			/>
			</View>		
		);
	}
	
}