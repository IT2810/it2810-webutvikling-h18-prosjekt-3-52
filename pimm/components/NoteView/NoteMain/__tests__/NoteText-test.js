import React from 'react';
import 'react-native';
import NoteText from '../NoteText.js';
import renderer from 'react-test-renderer';


test('NoteText renders correctly', () => {
  const titleRendered = renderer.create(<NoteText />).toJSON();
  expect(titleRendered).toMatchSnapshot();
  expect(titleRendered.type).toBe("View");
  expect(titleRendered.children[0].type).toBe("TextInput");
  expect(titleRendered.children[0].props.placeholder).toBe("Write note here...");
  
});

test('Change note', ()=>{
	const nt = renderer.create(<NoteText />).getInstance();
	expect(nt.state.note).toBe("");
	nt.changeNote("Original");
	expect(nt.state.note).toBe("Original");
});
