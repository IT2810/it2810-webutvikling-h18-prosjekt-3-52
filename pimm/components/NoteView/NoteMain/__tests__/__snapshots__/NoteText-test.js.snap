// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoteText renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
      "padding": 5,
    }
  }
>
  <TextInput
    allowFontScaling={true}
    multiline={true}
    onChangeText={[Function]}
    placeholder="Write note here..."
    style={
      Object {
        "backgroundColor": "beige",
        "padding": 5,
        "width": 280,
      }
    }
  />
</View>
`;
